src/NIST-800-53/rules/elasticache/nist80053ElasticacheRedisClusterAutomaticBackup.ts

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { CfnCacheCluster, CfnReplicationGroup } from '@aws-cdk/aws-elasticache';
+import { CfnResource } from '@aws-cdk/core';
+
+/**
+ * ElastiCache Redis clusters retain automatic backups for at least 15 days (Control IDs: CP-9(b), CP-10, SI-12)
+ * @param node the CfnResource to check
+ */
+export default function (node: CfnResource): boolean {
+  if (node instanceof CfnCacheCluster) {
+    const engine = node.engine.toLowerCase();
+    const retention = node.snapshotRetentionLimit;
+    if (engine == 'redis' && (retention == undefined || retention < 15)) {
+      return false;
+    }
+  }
+  if (node instanceof CfnReplicationGroup) {
+    if (node.engine != undefined) {
+      const engine = node.engine.toLowerCase();
+      const retention = node.snapshotRetentionLimit;
+      if (engine == 'redis' && (retention == undefined || retention < 15)) {
+        return false;
+      }
+    }
+  }
```

Comment:
In the case of a Replication Group, it's okay for the engine to be undefined as the value [can only be redis](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-replicationgroup.html#cfn-elasticache-replicationgroup-engine)

```suggestion
  else if (node instanceof CfnReplicationGroup) {
    const retention = node.snapshotRetentionLimit;
    if (retention == undefined || retention < 15) {
      return false;
    }
  }
```